cmd/cli/serve/serve.go

```diff
@@ -110,19 +119,40 @@ func serve(cmd *cobra.Command, cfg types.Bacalhau, fsRepo *repo.FsRepo) error {
 	ctx := cmd.Context()
 	cm := util.GetCleanupManager(ctx)
 
+	// Attempt to read the node name from the repo
 	nodeName, err := fsRepo.ReadNodeName()
 	if err != nil {
 		return fmt.Errorf("failed to get node name: %w", err)
 	}
 	if nodeName == "" {
-		nodeName, err = config.GenerateNodeID(ctx, cfg.NameProvider)
-		if err != nil {
-			return fmt.Errorf("failed to generate node name for provider %s: %w", cfg.NameProvider, err)
+		// name isn't set in repo, check if a flag was provided.
+		nodeName = cmd.PersistentFlags().Lookup(NameFlagName).Value.String()
+		if nodeName == "" {
+			// no flag provided, we need to generate it and persist
+			log.Info().Msgf("generating node name using provider %s", cfg.NameProvider)
+			nodeName, err = config.GenerateNodeID(ctx, cfg.NameProvider)
+			if err != nil {
+				return fmt.Errorf("failed to generate node name for provider %s: %w", cfg.NameProvider, err)
+			}
+			if err := fsRepo.WriteNodeName(nodeName); err != nil {
+				return fmt.Errorf("failed to write node name %s: %w", nodeName, err)
+			}
+			log.Info().Msgf("persisted node name %s", nodeName)
+		} else {
+			// the flag was provided with the name, persist it
+			log.Info().Msgf("persisting node name from flag %s", nodeName)
+			if err := fsRepo.WriteNodeName(nodeName); err != nil {
+				return fmt.Errorf("failed to write node name %s: %w", nodeName, err)
+			}
+			log.Info().Msgf("persisted node name %s", nodeName)
```

Comment:
nit: persisting node name logic is duplicated and can be improved. Also maybe too much logging

Reply:
done.